tools/env-loader/pkg/writers/dotenv.go

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright 2024 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package writers
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// Pulled from https://hexdocs.pm/dotenvy/dotenv-file-format.html#variable-names
+var keyValidationRegex = regexp.MustCompile("^[a-zA-Z_]+[a-zA-Z0-9_]*$")
+
+// Outputs the values in .env file format
+type DotenvWriter struct{}
+
+// Create a new dotenv-format writer
+func NewDotenvWriter() *DotenvWriter {
+	return &DotenvWriter{}
+}
+
+func (*DotenvWriter) validateValue(key, _ string) error {
```

Comment:
I think this function is small enough and only referenced once that it doesn't really need to exist. I would rather it just be inlined below.

Reply:
It's not clear from this PR (which I split tests out of) but because this approach is easier and cleaner to test. With the validation logic (which is currently very small) in a separate function, it makes tests both easier to write and easier to understand.

I'll inline it if you still want this - let me know.